sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp

```diff
@@ -857,2619 +857,2674 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
-                  (flushDataOptions.RetainUncommittedData.GetValue() ? "true" : "false")));
-        }
-        if (flushDataOptions.Close.HasValue())
-        {
-          request.GetUrl().AppendQueryParameter(
-              Details::c_QueryClose,
-              Storage::Details::UrlEncodeQueryParameter(
-                  (flushDataOptions.Close.GetValue() ? "true" : "false")));
-        }
-        if (flushDataOptions.ContentLength.HasValue())
-        {
-          request.AddHeader(
-              Details::c_HeaderContentLength,
-              std::to_string(flushDataOptions.ContentLength.GetValue()));
-        }
-        if (flushDataOptions.ContentMd5.HasValue())
-        {
-          request.AddHeader(Details::c_HeaderContentMd5, flushDataOptions.ContentMd5.GetValue());
-        }
-        if (flushDataOptions.LeaseIdOptional.HasValue())
-        {
-          request.AddHeader(
-              Details::c_HeaderLeaseIdOptional, flushDataOptions.LeaseIdOptional.GetValue());
-        }
-        if (flushDataOptions.CacheControl.HasValue())
-        {
-          request.AddHeader(
-              Details::c_HeaderCacheControl, flushDataOptions.CacheControl.GetValue());
-        }
-        if (flushDataOptions.ContentType.HasValue())
-        {
-          request.AddHeader(Details::c_HeaderContentType, flushDataOptions.ContentType.GetValue());
-        }
-        if (flushDataOptions.ContentDisposition.HasValue())
-        {
-          request.AddHeader(
-              Details::c_HeaderContentDisposition, flushDataOptions.ContentDisposition.GetValue());
-        }
-        if (flushDataOptions.ContentEncoding.HasValue())
-        {
-          request.AddHeader(
-              Details::c_HeaderContentEncoding, flushDataOptions.ContentEncoding.GetValue());
-        }
-        if (flushDataOptions.ContentLanguage.HasValue())
-        {
-          request.AddHeader(
-              Details::c_HeaderContentLanguage, flushDataOptions.ContentLanguage.GetValue());
-        }
-        if (flushDataOptions.IfMatch.HasValue())
-        {
-          request.AddHeader(Details::c_HeaderIfMatch, flushDataOptions.IfMatch.GetValue());
-        }
-        if (flushDataOptions.IfNoneMatch.HasValue())
-        {
-          request.AddHeader(Details::c_HeaderIfNoneMatch, flushDataOptions.IfNoneMatch.GetValue());
-        }
-        if (flushDataOptions.IfModifiedSince.HasValue())
-        {
-          request.AddHeader(
-              Details::c_HeaderIfModifiedSince, flushDataOptions.IfModifiedSince.GetValue());
-        }
-        if (flushDataOptions.IfUnmodifiedSince.HasValue())
-        {
-          request.AddHeader(
-              Details::c_HeaderIfUnmodifiedSince, flushDataOptions.IfUnmodifiedSince.GetValue());
-        }
-        if (flushDataOptions.ClientRequestId.HasValue())
-        {
-          request.AddHeader(
-              Details::c_HeaderClientRequestId, flushDataOptions.ClientRequestId.GetValue());
+  namespace Details {
```

Comment:
At a quick glance, all this seems reasonable. How come the change in datalake/shares was a simple addition of a single Details group around the rest client, but for blobs, we need a lot more sections, here: https://github.com/Azure/azure-sdk-for-cpp/pull/921

cc @Jinming-Hu

Reply:
Also, can this `Details` section get moved to the existing one above?
https://github.com/Azure/azure-sdk-for-cpp/blob/2fe5120fd4af9d884d78f7ff4fa0c7c80b720446/sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp#L24-L26

Reply:
> At a quick glance, all this seems reasonable. How come the change in datalake/shares was a simple addition of a single Details group around the rest client, but for blobs, we need a lot more sections, here: #921

storage team has two different generators.

Reply:
> Also, can this `Details` section get moved to the existing one above?
> https://github.com/Azure/azure-sdk-for-cpp/blob/2fe5120fd4af9d884d78f7ff4fa0c7c80b720446/sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp#L24-L26

I think this is a minor improvement in code structure, it is currently not an official generator so improvement like this will not be a high priority.